tests/large/test_web.py

```diff
@@ -115,6 +116,50 @@ def language_file_exists(self, language):
         return self.path_exists(os.path.join(self.installed_path, "dictionaries", "{}.aff".format(language)))
 
 
+class PhantomJSTests(LargeFrameworkTests):
+    """The default Go google compiler."""
+
+    TIMEOUT_INSTALL_PROGRESS = 300
+
+    EXAMPLE_PROJECT = """console.log('hello, world');
+                         phantom.exit();"""
+
+    def setUp(self):
+        super().setUp()
+        self.installed_path = os.path.join(self.install_base_path, "web", "phantomjs")
+        self.framework_name_for_profile = "PhantomJS"
+
+    @property
+    def exec_path(self):
+        return os.path.join(self.installed_path, "bin", "phantomjs")
+
+    def test_default_phantomjs_install(self):
+        """Install PhantomJS from scratch test case"""
+        if not self.in_container:
+            self.example_prog_dir = tempfile.mkdtemp()
+            self.additional_dirs.append(self.example_prog_dir)
+            example_file = os.path.join(self.example_prog_dir, "hello.js")
+            open(example_file, "w").write(self.EXAMPLE_PROJECT)
+            compile_command = ["bash", "-l", "-c", "phantomjs {}".format(example_file)]
+        else:  # our mock expects getting that path
+            compile_command = ["bash", "-l", "phantomjs /tmp/hello.js"]
+
+        self.child = spawn_process(self.command('{} web phantomjs'.format(UMAKE)))
+        self.expect_and_no_warn("Choose installation path: {}".format(self.installed_path))
+        self.child.sendline("")
+        self.expect_and_no_warn("Installation done", timeout=self.TIMEOUT_INSTALL_PROGRESS)
+        self.wait_and_close()
+
+        self.assert_exec_exists()
+        self.assertTrue(self.is_in_path(self.exec_path))
+
+        # compile a small project
+        output = subprocess.check_output(self.command_as_list(compile_command)).decode()\
+            .replace('\r', '').replace('\n', '')
```

Comment:
please, feel free to change wherever it makes sense :)
Just ensure this work both in a large tests and within the medium tests, as we don't have the exact same configuration there.